DownFall/Models/Board.swift

```diff
@@ -359,41 +340,39 @@ extension Board {
-            var updatedTiles = transformation.endTiles else { return Transformation.zero }
-        
-        if let pp = playerPosition,
-            case let .player(data) = updatedTiles[pp].type {
+            var updatedTiles = transformation.endTiles,
+            let pp = playerPosition,
+            case let .player(data) = updatedTiles[pp].type
+            else { return Transformation.zero }
             
-            var items = data.carry.items
-            items.append(item)
-            let newCarryModel = CarryModel(items: items)
-            let playerData = EntityModel(originalHp: data.originalHp,
-                                         hp: data.hp,
-                                         name: data.name,
-                                         // Orend
-                // we have to reset attack here because the player has moved but the turn may not be over
-                // Eg: it is possible that there could be two or more monsters
-                // under the player and the player should be able to attack
-                attack: data.attack.resetAttack(),
-                type: data.type,
-                carry: newCarryModel,
-                animations: data.animations,
-                abilities: data.abilities)
-            
-            updatedTiles[pp.x][pp.y] = Tile(type: .player(playerData))
-        }
+        let newCarryModel = data.carry.earn(item.amount, inCurrency: item.type.currencyType)
+        let playerData = EntityModel(originalHp: data.originalHp,
+                                     hp: data.hp,
+                                     name: data.name,
+                                     // Orend
+            // we have to reset attack here because the player has moved but the turn may not be over
+            // Eg: it is possible that there could be two or more monsters
+            // under the player and the player should be able to attack
+            attack: data.attack.resetAttack(),
+            type: data.type,
+            carry: newCarryModel,
+            animations: data.animations,
+            abilities: data.abilities)
+        
+        updatedTiles[pp.x][pp.y] = Tile(type: .player(playerData))
         
         tiles = updatedTiles
         
         return Transformation(transformation: transformation.tileTransformation,
-                              inputType: .collectItem(coord, item),
+                              inputType: .collectItem(coord, item, playerData.carry.total(in: item.type.currencyType)),
                               endTiles: updatedTiles)
     }
     
     
     private func monsterDied(at coord: TileCoord, input: Input) -> Transformation {
-        if case let .monster(monsterData) = tiles[coord].type,
-            let item = monsterData.carry.items.first {
+        if case let .monster(monsterData) = tiles[coord].type {
+            let gold = tileCreator.goldDropped(from: monsterData)
+            let item = Item.init(type: .gold, amount: gold)
```

Comment:
This doesn't let monsters drop other things.  Add some functionality to the entity so it gives you the item it drops or something

Reply:
This is specific to the item type.  Right now a monster can drop anything that it is carrying